### PR TITLE
Ss/pr wf refactoring

### PR DESCRIPTION
**Pull request by SiddhantSadangi**

### CODE REVIEW

File: .github/workflows/branch.yml [hunk 1]
```diff
@@ -52,28 +57,38 @@ jobs:
...
-          - integrations-and-supported-tools/matplotlib/scripts
-          - integrations-and-supported-tools/sklearn/scripts
-          - integrations-and-supported-tools/tensorflow-keras/scripts
-          - integrations-and-supported-tools/catalyst/scripts
-          - integrations-and-supported-tools/lightgbm/scripts
-          - integrations-and-supported-tools/xgboost/scripts
-          - integrations-and-supported-tools/pytorch-lightning/scripts
-          - integrations-and-supported-tools/fastai/scripts
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+        python-version: [ '3.7', '3.8', '3.9', '3.10' ]
```
Comment (kamil-kaczmarek) [inline .github/workflows/branch.yml:90]: for now let's do only for ubuntu and py 3.7, try how it works then expand

**Review by kamil-kaczmarek — CHANGES_REQUESTED** [review]

File: .github/workflows/branch.yml (hunk 1 above)
Comment (SiddhantSadangi) [reply .github/workflows/branch.yml:90]: I've actually tested on all combinations for both scripts and notebooks (hence the 50+ commits).
And since these will run only on the changed files, I think we can afford to include everything.
Worst case, we can have only the edge-cases (run only on python 3.7 and 3.10). This will half the number of tests.

What do you think?

Comment (kamil-kaczmarek) [reply .github/workflows/branch.yml:90]: @SiddhantSadangi ack

**Review by kamil-kaczmarek — APPROVED** [review]
